### DISCUSSION TITLE
Deprecated Annotation Not Reflected in Generated Client Code

### DISCUSSION THREAD

**Discussion by adiltirur** [serverpod/serverpod#2212] [General]
Hello everyone,

I've been working with Serverpod and I've come across an issue where the @Deprecated annotation in the server code is not being reflected in the generated client code.

Here's a snippet of the server code for context:
```
@Deprecated('Use getDistanceMatrixList instead')
Future<ApiDistanceMatrixResult?> getDistanceMatrix(
  Session session,
  String originLatitude,
  String originLongitude,
  String destination,
) async {
  final endPoint =
      _createDistanceMatrixUrl(originLatitude, originLongitude, destination);
  final response = await http.get(Uri.parse(endPoint));
  return _handleDistanceMatrixResponse(response);
}
```

In this snippet, the getDistanceMatrix method is marked as deprecated and the suggestion is to use getDistanceMatrixList instead. However, this deprecation warning is not showing up in the generated client code.

Has anyone else experienced this issue? Any suggestions on how to ensure that the @Deprecated annotation is carried over to the generated client code would be greatly appreciated.

Thank you in advance for your help.

**Comment by Isakdl** [comment]
This is currently not supported by the code generator but would be a nice addition. I made an issue from this discussion here: https://github.com/serverpod/serverpod/issues/2214

**Reply by adiltirur** [reply]
Thank you very much for opening the issues, this would be helpful for many devs. I appreciate the quick feedback.
